go/parser/parser.go

```diff
@@ -228,6 +237,48 @@ func (p *parser) redeclared(ident *ast.Ident, prev *ast.Object, reason string) {
 	p.error(ident.Pos(), fmt.Sprintf("%s redeclared%s in this block%s", ident.Name, reason, prevDecl))
 }
 
+var (
+	goosSuffixes = []string{
+		"_dragonfly.go",
+		"_netbsd.go",
+		"_openbsd.go",
+		"_solaris.go",
+		"_freebsd.go",
+		"_nacl.go",
+		"_android.go",
+		"_plan9.go",
+		"_darwin.go",
+		"_linux.go",
+		"_windows.go",
+	}
+	goarchSuffixes = []string{
+		"_amd64.go",
+		"_amd64p32.go",
+		"_arm.go",
+		"_ppc64.go",
+		"_ppc64le.go",
+		"_386.go",
+	}
+)
+
+func isPlatformDependent(f *token.File) bool {
```

Comment:
Yes...But now I haven't found a good way to handle this, because we use the go/build's default context which is for current GOOS and GOARCH. If we want all platform/arch associated files, we need pass all the combinations of GOOS/GOARCH/CGO... Any idea?